transport/src/main/java/io/netty/channel/pool/SimpleChannelPool.java

```diff
@@ -184,7 +184,7 @@ private void notifyHealthCheck(Future<Boolean> future, Channel ch, Promise<Chann
         assert ch.eventLoop().inEventLoop();
 
         if (future.isSuccess()) {
-            if (future.getNow() == Boolean.TRUE) {
+            if (future.getNow()) {
```

Comment:
@normanmaurer It converts Boolean to boolean. It is actually translated into: `future.getNow().booleanValue()`, which involves no boxing but just unboxing.